### PR TITLE
[launcher] Add configurable /dev/shm size and tmpfs mount option

### PR DESCRIPTION
**Pull request by alexmwu**
CS operators can use tee-dev-shm-size to increase the size of the /dev/shm mount for the workload.

CS operators can use `tee-mount` to specify tmpfs mounts in the running container. For example: `type=tmpfs,source=tmpfs,destination=/tmp;type=tmpfs,source=tmpfs,destination=/tmp2,size=12345`.
This interacts with the `tee.launch_policy.allow_mount_destination` launch policy that configures what the allowable mount points for a given workload are.

### CODE REVIEW

File: launcher/spec/launch_policy.go [hunk 1]
```diff
@@ -97,6 +105,25 @@ func GetLaunchPolicy(imageLabels map[string]string) (LaunchPolicy, error) {
 		}
 	}
 
+	// default is debug only for memoryMonitoring
+	if v, ok := imageLabels[memoryMonitoring]; ok {
+		launchPolicy.AllowedMemoryMonitoring, err = toPolicy(memoryMonitoring, v)
+		if err != nil {
+			return LaunchPolicy{}, fmt.Errorf("invalid image LABEL '%s'; contact the image author", memoryMonitoring)
+		}
+	}
```
Comment (jkl73) [inline launcher/spec/launch_policy.go:114]: repeated from above?

File: launcher/spec/launch_spec.go [hunk 2]
```diff
@@ -98,7 +107,10 @@ type LaunchSpec struct {
 	Hardened                   bool
 	MemoryMonitoringEnabled    bool
 	LogRedirect                LogRedirectLocation
-	Experiments                experiments.Experiments
+	Mounts                     []launchermount.Mount
+	// DevShmSize is specified in kiB.
+	DevShmSize  int64
```
[the remark this answers is not in the document]
Comment (alexmwu) [reply launcher/spec/launch_spec.go:112]: no point: https://pkg.go.dev/github.com/containerd/containerd/oci#WithDevShmSize

File: launcher/cloudbuild.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 substitutions:
   # using this base image for now, because there is an issue causing the newest COS dev
   # image not booting with cs.
-  '_BASE_IMAGE': '' # left empty means using the latest image in the family
+  '_BASE_IMAGE': 'cos-dev-117-18374-0-0' # left empty means using the latest image in the family
```
Comment (jkl73) [inline launcher/cloudbuild.yaml:4]: this is not needed anymore, and you probably need to rebase to the head

**Review by jkl73 — APPROVED** [review]

**Comment by alexmwu** [comment]
Merging past the failing lint, since it's on untouched code

**Pull request merged**